lib/reqTemplate.js

```diff
@@ -75,12 +71,32 @@ function prepareTemplateString(template, defaultLookupPath) {
  * @returns {Function} a template resolver for the provided template
  */
 function compileTemplate(templateSpec, setValue, reqPart) {
+    function createTAssemblyTemplate(template) {
+        if (/^(?:\$\.)?(?:[a-zA-Z][\w-]*)(?:\.[a-zA-Z][\w-]*)*$/.test(template)) {
```

Comment:
This RegExp could use some comments. At least what is it checking for. Ideally, though, each of its parts would be commented properly.